### DISCUSSION TITLE
Trouble Testing with EntitiesGraphQLQuery and EntitiesProjectionRoot

### DISCUSSION THREAD

**Discussion by san4d** [Netflix/dgs-framework#579] [Q&A]
# Setup
I have a federated GraphQL schema that looks something like this:
```graphql
type Parent @key(fields: "id") @extends {
    id: String! @external
    children(id: String, sort: Sort): [Child]
}

type Child {
    id: ID!
    name: String!
}
```

Based on the [federated testing](https://netflix.github.io/dgs/advanced/federated-testing/), I'm setting up my query like so (I'm using the generated client):
```java
var query = new EntitiesGraphQLQuery.Builder().addRepresentationAsVariable(
                            ParentRepresentation.newBuilder().id(UUID.random().toString()).build());
```

I then create and execute a request as follows:
```java
var request = new GraphQLQueryRequest(query, 
                              new EntitiesProjectionRoot().onParent().id().children().name());
var context = queryExecutor.executeAndGetDocumentContext(request.serialize(), query.getVariables());
var response = new GraphQLResponse(context.jsonString());
Parent p = response.extractValueAsObject("data._entities[0]", Parent.class);
```

# Tooling
- Java 11
- DGS SpringBoot Started 4.1.0
- DGS Platform Dependencies: 4.1.0

I'm going to upgrade the DGS dependencies and see if that helps.

# Problems 
## Extracting the Parent class
This line throws an error because of the returned `__typename` attribute. 
```
response.extractValueAsObject("data._entities[0], Parent.class);
```
Is there a way to ignore the attribute or have I misconfigured something (potentially the something in the codegen)?

## Passing in inputs to the child
My query should allow me to pass in the `id` and `sort` attributes. When I run it live behind a gateway, this behavior works as expected. However, it's not clear how to pass in the inputs to the fetcher for the child when using the generated client. I tried passing the `id` in along with the representations in the `executeAndGetDocumentContext`, where the provided variables looked something like this:
```json
{
  "representations": {
    "id": "a62d2ac5-525e-4472-90c4-50e8206b883a",
    "__typename": "Parent"
  },
  "id":  "c7a0842c-1422-4023-a84d-4167f2d0d888"
}
```
However, I don't see the child id passed in as a value in the `@DgsEntityFetcher` argument for the Parent or in the `DgsDataFetchingEnvironment` argument for the ` @DgsData(parentType = "Parent", field = "children")`. 

Thanks for your time!

**Comment by berngp** [comment]
Please upgrade dependencies, including the **Gradle DGS Codegen** plugin.

**Reply by san4d** [reply]
I'm using the Community Maven plugin as well. If there's a particular feature in the Gradle plugin that powers this behavior, please let me know. I can look for parity in the Maven plugin.
